zigzag/opt/loma/multipermute.py

```diff
@@ -69,10 +112,11 @@ def visit(h: ListElement) -> list[Any]:
     return this_list
 
 
-def permutations(multiset: list[Any]):
+def permutations(multiset: list[Any], constraints: list[PermutationConstraint]):
     """! Generator providing all multiset permutations of a multiset."""
     h, i, j = init(multiset)
-    yield visit(h)
+    if all([constr.is_valid(visit(h)) for constr in constraints]):
```

Comment:
```suggestion
    if all(constr.is_valid(visit(h)) for constr in constraints):
```

You can also do "list comprehension" without creating a list. I think this will also run a bit faster

Reply:
oops, didn't notice this was a different comment from the one underneath this one... sorry!